_source/_posts/2019-12-30-java-jersey-jaxrs.adoc

```diff
@@ -0,0 +1,595 @@
+
+Let's see how you can generate a valid token and how to add it to your request.
+
+== Generate a Valid Token
+
+To validate your request you need to add the `Authorization` header to the request. The header will provide the type of authentication and the token, which will look like the snippet below:
+
+[source,text]
+----
+-H 'Authorization: Bearer <token>'
+----
+
+To generate the token you can access https://oidcdebugger.com/ and provide the following information:
+
+- Authorize URI: `\https://{yourOktaDomain}/oauth2/default/v1/authorize`
+- Redirect URI: `\https://oidcdebugger.com/debug`
+- Client ID: `{yourClientID}`
+- Scope: `openid`
+- State: `dev`
+- Response type: `token`
+
+You can keep the already filled-in value for the "Nonce" field.
+
+After you fill in all the fields, click on **Send Request**. You'll be redirected to your Okta's App login page:
+
+image::{% asset_path 'blog/java-jersey-jaxrs/okta-login.png' %}[alt=Okta Login Page,width=400,align=center]
+
+Put your username and password, and click on **Sign In**. You'll be redirected to the OIDC Debugger again, where you'll see the generated token:
+
+image::{% asset_path 'blog/java-jersey-jaxrs/generated-token.png' %}[alt=OIDC Generated Token,width=800,align=center]
+
+Copy the value and replace with the `<token>` keyword in the command below:
+
+[source,text]
+----
+curl -X GET -o http://localhost:8080/tasks \
+  -H 'Authorization: Bearer <token>'
+----
+
+You'll see that the command now executes successfully:
+
+[source,text]
+----
+< HTTP/1.1 200
+< X-Content-Type-Options: nosniff
+< X-XSS-Protection: 1; mode=block
+< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
+< Pragma: no-cache
+< Expires: 0
+< X-Frame-Options: DENY
+< Content-Type: application/json
+< Content-Length: 2
+< Date: Thu, 21 Nov 2019 10:15:36 GMT
+<
+* Connection #1 to host localhost left intact
+[]
+----
+
+Let's register a task to make sure everything works as it should. Execute the following command into your terminal, replacing `<token>` by your token:
+
+[source,text]
+----
+curl -X POST \
+  http://localhost:8080/tasks \
+  -H 'Authorization: Bearer <token>' \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "description" : "Test my Jersey App!"
+  }'
+----
+
+Now let's execute the first command again:
+
+[source,text]
+----
+curl -X GET http://localhost:8080/tasks \
+  -H 'Authorization: Bearer <token>'
+----
+
+It now returns the task you just created!
+
+[source,json]
+----
+[{"id":"a44dba4f-d239-441a-925d-d9248aeb4925","description":"Test my Jersey App!"}]
+----
+
+Well done! You managed to create a CRUD service using Jersey! Even better, the service is secure and it took you minimal effort to make it happen.
+
+You can view the source code of this tutorial going to its https://github.com/oktadeveloper/jersey-rest-example[GitHub repository].
```

Comment:
Repo is missing the standard `okta-` prefix. Not sure it matters, but this is a convention we've been using for a while now. I drop the prefix when the example is not secured by Okta.